### PR TITLE
1/n Generalize internal checks for Accelerator in Trainer - remove trainer._device_type

### PR DESCRIPTION
**Pull request by four4fish**
## What does this PR do?
#11002 could be follow ups for this

Part of #10821

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1650,33 +1650,31 @@ def __setup_profiler(self) -> None:
...
                 "TPU available but not used. Set the `tpu_cores` flag in your trainer"
                 " `Trainer(tpu_cores=8)` or script `--tpu_cores=8`."
             )
 
-        if (
-            _IPU_AVAILABLE
-            and self._device_type != _AcceleratorType.IPU
-            and not isinstance(self.accelerator, IPUAccelerator)
-        ):
+        if _IPU_AVAILABLE and not isinstance(self.accelerator, IPUAccelerator):
```
Comment (ananthsub) [inline pytorch_lightning/trainer/trainer.py:1677]: i find this logging pretty verbose. For instance, I don't want to  see `"TPU available: False, using: 0 TPU cores"` on every run. This is only useful if there are TPUs available and I'm not using them. I assume that if there are not TPUs available but I have specified them, then an exception would be thrown before this is called.

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -1650,33 +1650,31 @@ def __setup_profiler(self) -> None:
...
-            self.tpu_cores if self.tpu_cores is not None and self._device_type == _AcceleratorType.TPU else 0
+            self.tpu_cores if self.tpu_cores is not None and isinstance(self.accelerator, TPUAccelerator) else 0
         )
         rank_zero_info(f"TPU available: {_TPU_AVAILABLE}, using: {num_tpu_cores} TPU cores")
 
         num_ipus = self.ipus if self.ipus is not None else 0
         rank_zero_info(f"IPU available: {_IPU_AVAILABLE}, using: {num_ipus} IPUs")
 
-        if torch.cuda.is_available() and self._device_type != _AcceleratorType.GPU:
+        if torch.cuda.is_available() and isinstance(self.accelerator, GPUAccelerator):
```
Comment (ananthsub) [inline pytorch_lightning/trainer/trainer.py:1665]: this is incorrect. the prior code was checking if the GPU accelerator was *not* being used.
But I don't think we need to check the accelerator instance at all. Why can't we check the `gpus` flag directly?

Same as above, I don't get why we have to check TPU Accelerator instead of `tpu_cores`

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/trainer/trainer.py:1677]: Fair point. What about when users pass `(accelerator="auto", devices="x")`?

Comment (justusschock) [reply pytorch_lightning/trainer/trainer.py:1677]: can we simply lower the logging level to maybe debug here? I also find it to verbose most of the times.

@kaushikb11 same goes here. Usually people know on what kind of machine they submitted their job.

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (four4fish) [reply pytorch_lightning/trainer/trainer.py:1665]: With ppl reach agreement on gpu/cpus vs devices in issue #10410 Future of gpus/ipus/tpu_cores with respect to devices, I think check accelerator and device number will be better than tpu/gpus.

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (four4fish) [reply pytorch_lightning/trainer/trainer.py:1677]: Totally agree it's too verbose, and why do we have _log_device_info() with all the warnings in trainer? Should we move _log_device_info to the end of the accelerator_connector init, or having the warning relocate to select functions in accelerator connector?

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (tchaton) [inline pytorch_lightning/trainer/trainer.py:1665]:
```suggestion
        if torch.cuda.is_available() and not isinstance(self.accelerator, GPUAccelerator):
```

File: pytorch_lightning/trainer/trainer.py [hunk 3]
```diff
@@ -1650,33 +1650,31 @@ def __setup_profiler(self) -> None:
...
             rank_zero_warn(
                 "GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.",
                 category=PossibleUserWarning,
             )
 
-        if _TPU_AVAILABLE and self._device_type != _AcceleratorType.TPU:
+        if _TPU_AVAILABLE and isinstance(self.accelerator, TPUAccelerator):
```
Comment (tchaton) [inline pytorch_lightning/trainer/trainer.py:1671]:
```suggestion
        if _TPU_AVAILABLE and not isinstance(self.accelerator, TPUAccelerator):
```

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (four4fish) [reply pytorch_lightning/trainer/trainer.py:1677]: created a issue #11014 to discuss and will address it in a different PR
